### PR TITLE
SNOW-1325652: release snowpark 1.15.0

### PR DESCRIPTION
**Pull request by sfc-gh-aling**
Please answer these questions before submitting your pull requests. Thanks!

1. What GitHub issue is this PR addressing? Make sure that there is an accompanying issue to your PR.

   Fixes #NNNN

2. Fill out the following pre-review checklist:

   - [ ] I am adding a new automated test(s) to verify correctness of my new code
   - [ ] I am adding new logging messages
   - [ ] I am adding a new telemetry message
   - [ ] I am adding new credentials
   - [ ] I am adding a new dependency

3. Please describe how your code solves the related issue.

   Please write a short description of how your code change solves the related issue.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,6 +1,6 @@
 # Release History
 
-## 1.15.0 (TBD)
+## 1.15.0 (2024-04-24)
 
 ### New Features
```
Comment (sfc-gh-mabrennan) [inline CHANGELOG.md:6]: Added support for registering udfs and stored procedure to local testing.
->
Added support for registering UDFs and stored procedures to local testing.

Comment (sfc-gh-mabrennan) [reply CHANGELOG.md:6]: Added distributed tracing using open telemetry apis
->
 Added distributed tracing using open telemetry APIs

Comment (sfc-gh-mabrennan) [reply CHANGELOG.md:6]: Re: "UDAF client support is ready for public preview. Please stay tuned for the Snowflake announcement of UDAF public preview." Please discuss with PM if they want forward-looking statements in the documentation.

**Review by sfc-gh-mabrennan — APPROVED** [review]
Changelog looks good. I made a few small suggestions. Thanks!

File: CHANGELOG.md (hunk 1 above)
Comment (sfc-gh-sfan) [reply CHANGELOG.md:6]: cc @sfc-gh-jfreeberg ^

Comment (sfc-gh-jfreeberg) [reply CHANGELOG.md:6]: Yeah, the server changes will be available on May 6th. Public blog post on May 8. I'm OK with this language

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,11 +1,45 @@
...
+  - snowflake.snowpark.DataFrameWriter:
+    - save_as_table
+- Added support for snow:// URLs to `snowflake.snowpark.Session.file.get` and `snowflake.snowpark.Session.file.get_stream`
+- Added support to register stored procedures and UDxFs with a `comment`.
+- UDAF client support is ready for public preview. Please stay tuned for the Snowflake announcement of UDAF public preview.
+- Added support for dynamic pivot.  This feature is currently in private preview.
+
+### Improvements
+
+- Improved the generated query performance for both compilation and execution by converting duplicate subqueries to Common Table Expressions (CTEs). It is still an experimental feature, and can be enabled by setting `session.cte_optimization_enabled` to `True`.
```
Comment (sfc-gh-jdu) [inline CHANGELOG.md:29]:
```suggestion
- Improved the generated query performance for both compilation and execution by converting duplicate subqueries to Common Table Expressions (CTEs). It is still an experimental feature not enabled by default, and can be enabled by setting `session.cte_optimization_enabled` to `True`.
```

**Review by sfc-gh-jdu — APPROVED** [review]

**Review by sfc-gh-jrose — APPROVED** [review]

**Review by sfc-gh-stan — APPROVED** [review]

**Review by sfc-gh-mabrennan — APPROVED** [review]
The updates to the changelog look good. Thanks!

**Pull request merged**